src/Data/String/CodePoints.purs

```diff
@@ -143,19 +176,41 @@ countTail p s accum = case uncons s of
 -- | Drops the given number of code points from the beginning of the string. If
 -- | the string does not have that many code points, returns the empty string.
 -- | Operates in constant space and in time linear to the given number.
+-- |
+-- | ```purescript
+-- | drop 5 "𝐀𝐀 b c"
+-- |    == "c"
+-- | -- compared to Data.String:
+-- | drop 5 "𝐀𝐀 b c"
+-- |    == "b c" -- because "𝐀" is 2 chars long
+-- | ```
+-- |  
 drop :: Int -> String -> String
 drop n s = String.drop (String.length (take n s)) s
 
 
 -- | Drops the leading sequence of code points which all match the given
 -- | predicate from the string. Operates in constant space and in time linear
 -- | to the length of the string.
+-- |
+-- | ```purescript
+-- | dropWhile (\c -> (_ == 0x1D400) $ codePointToInt c) "𝐀𝐀 b c 𝐀"
+-- |    == " b c 𝐀" 
+-- | ```
+-- |
 dropWhile :: (CodePoint -> Boolean) -> String -> String
 dropWhile p s = drop (count p s) s
 
 
 -- | Creates a string from an array of code points. Operates in space and time
 -- | linear to the length of the array.
+-- |
+-- | ```purescript
+-- | codePointArray = toCodePointArray "b c 𝐀"
```

Comment:
If you want to avoid `Maybe` you can only use `codePointFromChar` or `toCodePointArray`. Since all other forms for creating a `CodePoint` return a `Maybe CodePoint`. (AFAIK)

another option would be:
```purescript
>>> codePointArray = toCodePointArray "c 𝐀"
>>> codePointArray
[CodePoint 0x63, CodePoint 0x20, CodePoint 0x1D400]
>>> fromCodePointArray codePointArray
"c 𝐀"
```
This is also not optinal, since the first two lines basically explain `toCodePointArray`

with `codePointFromChar`:

```purescript
>>> codePointArray = [codePointFromChar 'A', codePointFromChar ' ']
>>> fromCodePointArray codePointArray
"A "
```
Disadvantage of this version: It does not show what CodePoint can do that Char can't.

Reply:
There is the same problem with `codePointToInt`:
```purescript
>>> map codePointToInt (codePointFromInt 0x1D400)
Just  0x1D400
```

Reply:
Right, yeah. I think I like this option the best:

```
>>> codePointArray = toCodePointArray "c 𝐀"
>>> codePointArray
[CodePoint 0x63, CodePoint 0x20, CodePoint 0x1D400]
>>> fromCodePointArray codePointArray
"c 𝐀"
```

Reply:
I think in most actual uses of this function, you'll be doing something like `toCodePointArray >>> stuff >>> fromCodePointArray`, and constructing an array of code points with `toCodePointArray` is not really any worse than constructing one with `traverse codePointFromInt`, in retrospect.

Reply:
and for `codePointToInt`?
```purescript
>>> boldA = codePointFromInt 0x1D400
>>> boldA
Just (CodePoint 0x1D400)
>>> map codePointToInt boldA

Reply:
I think using `map` is probably the best approach there. Perhaps we could use a BMP example first though, together with `codePointFromChar`, so that we have a simpler example which doesn't involve `Maybe`?